### PR TITLE
EOD summary script added

### PR DESCRIPTION
**Pull request by AndrewClugston**
This script can be used to produce a summary at the end of day which includes the volume traded, total amount of trades, average spread and time weighted average spread by sym.  This script should be set on a crontab to run at the end of day.

### CODE REVIEW

**Comment by ThomasSmyth** [comment]
This has been open a while, anything happening here?

File: code/rdb/eodsummary.q [hunk 1]
```diff
@@ -0,0 +1,62 @@
+
+\d .eodsum
+rdbtypes:@[value;`rdbtypes;`rdb];                                                               //list of rdb types to look for and call in rdb
+rdbnames:@[value;`rdbnames;()];                                                                 //list of rdb names to search for and call in rdb
+schema:@[value;`schema;1b];                                                                     //retrieve the schema from the rdb
+rdbconnsleepintv:@[value;`rdbconnsleepintv;10];                                                 //number of seconds between attempts to connect to the rdb
+
+if[not .timer.enabled;.lg.e[`symcheckinit;
```
Comment (ThomasSmyth) [inline code/rdb/eodsummary.q]: Move entire `.lg.e[...` use to next line.

File: code/rdb/eodsummary.q [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+   "the timer must be enabled to run the symcheck process"]];                                   // if the timer is not enabled, then exit with error
```
Comment (ThomasSmyth) [inline code/rdb/eodsummary.q]: Newline for closing bracket.

File: code/rdb/eodsummary.q [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+.servers.CONNECTIONS:distinct .servers.CONNECTIONS,.eodsum.rdbtypes;                            // make sure that the process will make a connection to any process of rdb type
+.lg.o[`init;"searching for servers"];                                                           //and append connection results to logfile
+
+.servers.startup[];
+
+.eodsum.subscribe[]                                                                             //subscribe to the rdb
+
+while[
+  .eodsum.nordbconnected[];                                                                     // check if the rdb has connected, block the process until connection is established
+  .os.sleep[.eodsum.rdbconnsleepintv];                                                          // while not connected, proc sleeps for X seconds then runs the subscribe function again
```
Comment (ThomasSmyth) [inline code/rdb/eodsummary.q]: Trim squares.

File: code/rdb/eodsummary.q [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+createsummary:{
+  update `$string bucket from `.eodsum.c;                                                       //change type from long to sym
+  d:exec distinct bucket from .eodsum.c;                                                        //get all unique values to be used as column headers
+  twas:exec d#(bucket!twas) by sym:sym from .eodsum.c;                                          //pivot table
+  `summarytab set 0!(uj/)(.eodsum.voltrd;.eodsum.avgsprd;twas);
+ };
+
+createsummary[];
+
+savepath:hsym `$":",getenv[`TORQHOME],"/hdb/database/";
```
Comment (ThomasSmyth) [inline code/rdb/eodsummary.q]: Define this at the start of the file with the rest of the vars (before `\d .eodsum` if needs be) + I think this path should be config driven.

Comment (jonathonmcmurray) [reply code/rdb/eodsummary.q]: I think this is also wrong - there should be a `KDBHDB` or `TORQHDB` env var which you should be using, especially significant as your setup does NOT use the `hdb` dir under `TORQHOME`, so where you're saving here will never get loaded

**Comment by ThomasSmyth** [comment]
There is inconsistent indentation of actions following `if` statements + spacing in comments.

**Comment by AndrewClugston** [comment]
Moved to TorQ FSP

**Pull request closed without merging**